jekyll/_cci2/build-publish-snap-packages.md

```diff
@@ -0,0 +1,176 @@
+
+## Build Environment
+
+```
+...
+version: 2
+jobs:
+  build:
+    docker:
+      - image: cibuilds/snapcraft:stable
+...
+```
+
+The `docker` executor is used here with the `cibuilds/snapcraft` Docker image. This image includes the `snapcraft` command which will be used to build the actual snap. This image is based on the official `snapcore/snapcraft` Docker image by Canonical with all of the command-line tools you'd want to be installed in a CI environment. The source for `cibuilds/snapcraft` can be found [here][cibuilds-snapcraft] and the source for the upstream image [here][snapcore-snapcraft].
+
+## Running Snapcraft
+
+To build a snap in any environment (local, company servers CI, etc) there needs to be a Snapcraft config file. Typically this will be located at `snap/snapcraft.yml`. This doc assumes you already have this file and can build snaps successfully on your local machine. If not, you can read through the [Build Your First Snap][snap-guide-1] doc by Snapcraft to get your snap building on your local machine.
+
+```
+...
+    steps:
+      - checkout
+      - run:
+          name: "Build Snap"
+          command: snapcraft
+...
+```
+
+On CircleCI, this single command is needed to actually build your snap. This will run Snapcraft, which will then go through all of it's build steps and generate a `.snap` file for you. This file will typically be in the format of `<snap-name>-<snap-version>-<system-arch>.snap`.
+
+## Testing
+
+Unit testing your software's code has been covered extensively in [our blog][blog], [our docs][docs], and around the Internet. Searching for your language/framework and unit testing or CI will turn up tons of information.
+
+You'll likely want to create a `job` before building the snap that pull project dependencies, any pre-checks you'd want to do, testing, and compiling.
+
+Building snaps on CircleCI means we end with a `.snap` file which we can test in addition to the code that created it. How you test the snap itself is up to you. Some users will attempt to install the snap in various distros and then run a command to make sure that installation process works. Snapcraft offers a build fleet for spreadtesting that allows you to test snaps on different distros, after you've already tested the code itself. This can be found [here][snapcraft-build].
+
+## Publishing
+
+Publishing a snap is more or less a two-step process. The first step which is done once, is to create a Snapcraft "login file" on your local machine that we upload to CircleCI. Assuming your local machine already has these tools installed and you are logged in to the Snapcraft Store (`snapcraft login`), we use the command `snapcraft export-login snapcraft.login` to generate a login file called `snapcraft.login`. As we don't want this file visible to the public or stored in the Git repository, we will base64 encode this file and store it in a [private environment variable][private-envars] called `$SNAPCRAFT_LOGIN_FILE`.
```

Comment:
Consider commenting the actual commands we're punching into the terminal? I know there's no real good way to do this, but worth thinking about.

An alternative is to just put the commands at the top and explain them later. Shoot first and answer questions later.

Another comment on this paragraph is that you're clearly calling out a "two-step process", but it's visually hard to see what those two steps are. Consider using a numbered list for those steps?